### PR TITLE
feat: overhaul person view

### PR DESCRIPTION
**Pull request by jhackett1**
this overhauls the person/resident view with:
- a fresher ui that fits more on the screen at once, making better use of space
- adds an overview page that presents a snapshot of a case
- broken the timeline down into an activity feed, case note list and workflow chain, to better suit the way people work

adds new interface patterns:
- inline editing
- slightly adapted `<Collapsible/>`s
- workflow chain/tree

most of the interesting stuff is in:
- `/pages/residents/[id]`
- `/components/ResidentPage`

builds on research from amy, elsa, ethan, gill, ruksana.

also adds two new env variables:
```
NEXT_PUBLIC_GOOGLE_CLIENT_KEY=
NEXT_PUBLIC_CORE_PATHWAY_APP_URL=https://workflows-social-care-service-staging.hackney.gov.uk
```

we are attempting to address these scenarios and hypotheses: https://www.figma.com/file/rsoEh303EFv41OuqORa6Sp/Product-strategy?node-id=9474%3A20462

figma design reference: https://www.figma.com/file/rsoEh303EFv41OuqORa6Sp/Product-strategy?node-id=9294%3A20301

### CODE REVIEW

File: components/CaseStatus/CaseStatusFlag/CaseStatusFlag.tsx [hunk 1]
```diff
@@ -9,51 +8,44 @@ interface Props {
   person: Resident;
 }
 
-const CaseStatusFlag = ({ person }: Props): React.ReactElement => {
-  const { data, error } = useCaseStatuses(person.id);
+const CaseStatusFlag = ({ person }: Props): React.ReactElement | null => {
+  const { data } = useCaseStatuses(person.id);
 
-  if (error) {
```
Comment (jhackett1) [inline components/CaseStatus/CaseStatusFlag/CaseStatusFlag.tsx:15]: we don't care about errors here, let it fail silently

File: components/ErrorMessage/ErrorMessage.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ const ErrorMessage = ({
   label = 'There was a problem. Please refresh the page or try again later.',
   className,
 }: Props): React.ReactElement => (
-  <span className={cx('govuk-error-message', className)}>
```
Comment (jhackett1) [inline components/ErrorMessage/ErrorMessage.tsx:12]: make styles consistent

File: components/Relationships/RemoveRelationshipDialog/RemoveRelationshipDialog.tsx [hunk 3]
```diff
@@ -38,7 +38,7 @@ const RemoveRelationshipDialog = ({
           href="#"
           onClick={onDismiss}
         >
-          Cancel
+          No, do nothing
```
Comment (jhackett1) [inline components/Relationships/RemoveRelationshipDialog/RemoveRelationshipDialog.tsx:41]: consistent wording

**Review by acodeninja — CHANGES_REQUESTED** [review]
It's great that you've added a test for the local storage adapter stuff, but really we need tests adding to this PR. Even if you just put happy path tests together for the pages, and more complete tests for the re-usable components like `Collapsible` or `InlineEditor` that would be great. Feel free to use the `.toMatchSnapshot()` jest stuff to reduce the amount of work this will involve.

File: pages/residents/[id]/index.tsx [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+import SEXUAL_ORIENTATIONS from 'data/orientation';
+import languages from 'data/languages';
+import WorkflowTree from 'components/ResidentPage/WorkflowTree';
+import useWorkflows from 'hooks/useWorkflows';
+
+interface Props {
+  resident: Resident;
+}
+
+// const getMostRecentChain = (workflows: Workflow[]): Workflow[] => {
```
[the remark this answers is not in the document]
Comment (jhackett1) [reply pages/residents/[id]/index.tsx:22]: this is still wip

**Comment by jhackett1** [comment]
i'm intending to add tests to _everything_

**Review by mapleleafjack — CHANGES_REQUESTED** [review]
Hey @jhackett1, the PR per se looks good (as the code is clean / well split / tests / ecc)

I've found one UI issue that I would like to see fixed:
Whenever you open any modal window in the new layout (clicking "see all workflows / fields ecc) the page messes up its layout by moving everything a couple of pixel on the side. This is - at least in my opinion - a bit annoying- Can I ask you to fix this? (let me know and we can pair on this in case)

**Comment by jhackett1** [comment]
> Hey @jhackett1, the PR per se looks good (as the code is clean / well split / tests / ecc)
> 
> I've found one UI issue that I would like to see fixed: Whenever you open any modal window in the new layout (clicking "see all workflows / fields ecc) the page messes up its layout by moving everything a couple of pixel on the side. This is - at least in my opinion - a bit annoying. Can I ask you to fix this? (let me know and we can pair on this in case)

fixed: https://github.com/LBHackney-IT/lbh-social-care-frontend/pull/834/commits/8c85b24704ff72d73a8c9eedecec9603f66e6115

**Review by mapleleafjack — APPROVED** [review]

**Comment by rs2416** [comment]
There's a small issue when I try the inplace editing the address in the housing section. The incorrect text shows in the modaI rather than the address.
Screenshot 2022-02-04 at 15 54 47

**Pull request merged**